base/loading.jl

```diff
@@ -1683,21 +1683,23 @@ function create_expr_cache(pkg::PkgId, input::String, output::String, concrete_d
     end
     deps_eltype = sprint(show, eltype(concrete_deps); context = :module=>nothing)
     deps = deps_eltype * "[" * join(deps_strs, ",") * "]"
-    trace = isassigned(PRECOMPILE_TRACE_COMPILE) ? `--trace-compile=$(PRECOMPILE_TRACE_COMPILE[])` : ``
-    io = open(pipeline(`$(julia_cmd()::Cmd) -O0
-                       --output-ji $output --output-incremental=yes
-                       --startup-file=no --history-file=no --warn-overwrite=yes
-                       --color=$(have_color === nothing ? "auto" : have_color ? "yes" : "no")
-                       $trace
-                       -`, stderr = internal_stderr, stdout = internal_stdout),
-              "w", stdout)
-    # write data over stdin to avoid the (unlikely) case of exceeding max command line size
-    write(io.in, """
+    withenv("OPENBLAS_NUM_THREADS" => 1) do
```

Comment:
Instead of using withenv, let’s make use of addenv on the process object, so that we aren’t flipping our own environment, which is generally a thread-unsafe operation.

Reply:
👍

Reply:
Updated to use `addenv`